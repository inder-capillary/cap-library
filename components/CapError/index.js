/**
*
* CapLabel
*
*/

import styled from 'styled-components';
import PropTypes from 'prop-types';
import { FONT_SIZE_S, CAP_RED, CAP_ORANGE } from '../styled/variables';

const types = {
  error: {
    color: `${CAP_RED}`,
  },
  warning: {
    color: `${CAP_ORANGE}`,
  },
};
const CapError = styled.div`
  font-size: ${FONT_SIZE_S};
  font-weight: normal;
  color: ${[(props) => types[props.type].color]};
  line-height: 20px;
`;

const CapErrorInline = styled.span`
  font-size: ${FONT_SIZE_S};
  font-weight: normal;
  color: ${[(props) => types[props.type].color]};
  line-height: 20px;
`;

CapError.CapErrorInline = CapErrorInline;

CapError.propTypes = {
  type: PropTypes.string,
};

CapError.defaultProps = {
  type: 'error',
};

CapErrorInline.propTypes = {
  type: PropTypes.string,
};

CapErrorInline.defaultProps = {
  type: 'error',
};

export default CapError;
