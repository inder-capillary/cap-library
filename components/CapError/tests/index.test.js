import React from 'react';
import renderer from 'react-test-renderer';
import CapError from '../index';

describe('<CapError />', () => {
  it('CapLabel renders correctly', () => {
    const tree = renderer.create(<CapError />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
