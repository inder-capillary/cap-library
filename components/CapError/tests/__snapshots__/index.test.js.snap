// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CapError /> CapLabel renders correctly 1`] = `
<div
  className="sc-bdVaJa cnvnJd"
  type="error"
/>
`;
